### PR TITLE
websocketpp: patch in build() instead of source() + bump deps

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **websocketpp/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by Croydon — APPROVED** [review]

File: recipes/websocketpp/all/conanfile.py [hunk 1]
```diff
@@ -20,28 +19,24 @@ def _source_subfolder(self):
...
-        self._patch_sources()
-
-    def _patch_sources(self):
-        if "patches" in self.conan_data and self.version in self.conan_data["patches"]:
-            for patch in self.conan_data["patches"][self.version]:
-                tools.patch(**patch)
 
     def package(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
```
Comment (SpaceIm) [inline recipes/websocketpp/all/conanfile.py:36]:
```suggestion

    def build(self):
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)

    def package(self):
```

**Review by SSE4 — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

File: recipes/websocketpp/all/conanfile.py [hunk 2]
```diff
@@ -20,28 +19,26 @@ def _source_subfolder(self):
...
-                tools.patch(**patch)
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
 
     def package(self):
         self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
         # We have to copy the headers manually, since the upstream cmake.install() step doesn't do so.
-        self.copy(pattern="*.hpp", dst="include/websocketpp", src=self._source_subfolder + "/websocketpp")
+        self.copy(pattern=os.path.join("websocketpp","*.hpp"), dst="include", src=self._source_subfolder)
```
Comment (SpaceIm) [inline recipes/websocketpp/all/conanfile.py:41]:
```suggestion
        self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, "websocketpp"))
```

feels more intuitive to me, but maybe it's just an habit.

Comment (prince-chrismc) [reply recipes/websocketpp/all/conanfile.py:41]: I did not like that it needed `websocketpp` directory twice

Comment (SpaceIm) [reply recipes/websocketpp/all/conanfile.py:41]: Ok I was disturbed by folder in the pattern, it's equivalent.

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**